x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizer.java

```diff
@@ -50,20 +52,31 @@ protected List<Batch<LogicalPlan>> batches() {
         rules.add(local);
         // TODO: if the local rules haven't touched the tree, the rest of the rules can be skipped
         rules.addAll(asList(operators(), cleanup()));
-        replaceRules(rules);
-        return rules;
+        return replaceRules(rules);
     }
 
+    @SuppressWarnings("unchecked")
     private List<Batch<LogicalPlan>> replaceRules(List<Batch<LogicalPlan>> listOfRules) {
```

Comment:
`listOfBatches` I'd say.

Reply:
This was unchanged, but yes, good point. I'll try to pick this up with a next update, I'd like to try adding [a verification](https://github.com/elastic/elasticsearch/pull/115858/files#r1836931039) after the local plan optimisation too.